Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -1010,8 +1043,10 @@ void GraphicsLayerCoordinated::commitLayerChanges(CommitState& commitState, floa
     if (m_pendingChanges.contains(Change::ContentsClippingRect))
         m_platformLayer->setContentsClippingRect(m_contentsClippingRect);
 
+    updateRootRelativeScale(); // Needs to happen before Change::ContentsScale.
```

Comment:
I am not too familiar with this yet, but I understand how we need the "side effect" of the `notLayerPropertyChanged` before we can proceed.

Would it make sense to return `m_rootRelativeScaleFactor` and use the returned value in line 1049, rather than the member variable? I feel that would make the code slightly more robust against refactoring.

Reply:
We need to cache the root relative scale factor somewhere, so we can return it if m_shouldUpdateRootRelativeScaleFactor=false, and we want to avoid recomputing it, since it's not cheap.

Or did I misunderstand your proposal?

Reply:
Yes, I am OK with the behaviour inside the function (`updateRootRelativeScale()`) itself, so I agree with caching the factor in a member variable.

My point is that rather than _using_ the member variable in this function, which means we 1) need to call the function that updates it 2) we need to use the member, and we have an _implicit_ linear dependency between the two (e.g. "before you can use this property, update it"), I think it's better if we instead do `auto scale = updateRootRelativeScale()`, then use `scale`. 

As of today, there would be no functional difference. But in an hypothetical refactoring down the line, having scale into a member variable initialised by the method is clearer, as they two need to remain tied together.

I hope it makes sense :-)